### ISSUE TITLE
focus trap for BModals doesn't work anymore since v0.23.5

### ISSUE THREAD

**Issue by tbl0605** [bootstrap-vue-next/bootstrap-vue-next#2064]
### Describe the bug

Hi,
the focus trap for BModals no longer works since v0.23.5
Or is it disabled by default now?

Reverting to version v0.23.4 solves the issue.

Reproduction:
- git clone https://github.com/tbl0605/boostrap-vue-next-bug
- npm install
- npm run serve
- click on button "Open Modal"
- use tab key

### Reproduction

https://github.com/tbl0605/boostrap-vue-next-bug

### Used Package Manager

npm

### Validations

- [X] Have tested with the latest version. This is still alpha version and sometime things change rapidly.
- [X] Follow our [Code of Conduct](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by tbl0605** [comment]
Hi @VividLemon,
thank you for your patch, focus trap seem to work again! ^^

I still have a question:
why did you add `tabindex="0"` in [BModal.vue](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/e63c230318dd24334b9700dbe1b60fa1974fe6a2/packages/bootstrap-vue-next/src/components/BModal/BModal.vue#L24)?

Focus traps should work without it no?
And the visual effect is pretty bad if you use `<b-modal centered />` for example:

![image](https://github.com/user-attachments/assets/bdf39344-3cf0-4a3c-b3dc-3713f81439ec)

I updated https://github.com/tbl0605/boostrap-vue-next-bug so you can see what I mean.

Should I open a new issue?
Thank you!
